kolena/_experimental/object_detection/evaluator.py

```diff
@@ -0,0 +1,367 @@
+        self,
+        test_case: TestCase,
+        inferences: List[Tuple[TestSample, GroundTruth, Inference]],
+        configuration: Optional[ThresholdConfiguration] = None,
+    ) -> List[Tuple[TestSample, TestSampleMetrics]]:
+        assert configuration is not None, "must specify configuration"
+        # compute thresholds to cache values for subsequent steps
+        self.compute_and_cache_f1_optimal_thresholds(configuration, inferences)
+        labels = {gt.label for _, gts, _ in inferences for gt in gts.bboxes}
+        return [(ts, self.compute_image_metrics(gt, inf, configuration, labels)) for ts, gt, inf in inferences]
+
+    def compute_aggregate_label_metrics(
+        self,
+        matchings: List[MulticlassInferenceMatches],
+        label: str,
+        thresholds: Dict[str, float],
+    ) -> ClassMetricsPerTestCase:
+        match_matched = []
+        match_unmatched_gt = []
+        match_unmatched_inf = []
+        confused_counter = 0
+        samples_with_label = 0
+        # filter the matching to only consider one class
+        for match in matchings:
+            sample_flag = False
+            for gt, inf in match.matched:
+                if gt.label == label:
+                    sample_flag = True
+                    match_matched.append((gt, inf))
+            for gt, inf in match.unmatched_gt:
+                if gt.label == label:
+                    sample_flag = True
+                    match_unmatched_gt.append((gt, inf))
+            for inf in match.unmatched_inf:
+                if inf.label == label:
+                    sample_flag = True
+                    match_unmatched_inf.append(inf)
+            if sample_flag:
+                samples_with_label += 1
+
+        all_bbox_matches_for_one_label = [
+            MulticlassInferenceMatches(
+                matched=match_matched,
+                unmatched_gt=match_unmatched_gt,
+                unmatched_inf=match_unmatched_inf,
+            ),
+        ]
+
+        tp = [inf for _, inf in match_matched if inf.score >= thresholds[inf.label]]
+        fp = [inf for inf in match_unmatched_inf if inf.score >= thresholds[inf.label]]
+        fn = [gt for gt, _ in match_unmatched_gt] + [
+            gt for gt, inf in match_matched if inf.score < thresholds[inf.label]
+        ]
+        tp_count = len(tp)
+        fp_count = len(fp)
+        fn_count = len(fn)
+        precision = compute_precision(tp_count, fp_count)
+        recall = compute_recall(tp_count, fn_count)
+        f1_score = compute_f1_score(tp_count, fp_count, fn_count)
+
+        average_precision = 0.0
+        if precision > 0:
+            baseline_pr_curve = compute_pr_curve(all_bbox_matches_for_one_label)
+            if baseline_pr_curve is not None:
+                average_precision = compute_average_precision(baseline_pr_curve.y, baseline_pr_curve.x)
+
+        return ClassMetricsPerTestCase(
+            Class=label,
+            TestSamples=samples_with_label,
```

Comment:
We might need to revisit this later once we have the multi-model support for nested metrics.  I think it makes more sense to use # of objects for statistical significance calculation for OD.

Reply:
Good thought - I'll add this too! Just thought it'd be good to know how many images are relevant to a particular class in a test case.